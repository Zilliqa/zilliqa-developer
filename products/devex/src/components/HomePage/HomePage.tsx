import React, { useContext } from 'react'
import { Spinner } from 'react-bootstrap'

import { NetworkContext } from 'src/services/network/networkProvider'

import Dashboard from './Dashboard/Dashboard'
import Searchbar from './Searchbar/Searchbar'

/*
            Home Layout
    ++++++++++++++++++++++++++++
    |        Search Bar        |
    ++++++++++++++++++++++++++++
    |                          |
    |                          |
    |        Dashboard         |
    |                          |
    |                          |
    ++++++++++++++++++++++++++++
*/
const HomePage: React.FC = () => {

  const networkContext = useContext(NetworkContext)
  const { isIsolatedServer } = networkContext!

  return (
    <>
      {isIsolatedServer !== null // wait for isolated server check to complete
        ? <div>
          <Searchbar isISSearchbar={isIsolatedServer} isHeaderSearchbar={false} />
          <Dashboard />
        </div>
        : <div className='center-spinner'><Spinner animation="border" /></div>
      }
    </>
  )
}

export default HomePage
