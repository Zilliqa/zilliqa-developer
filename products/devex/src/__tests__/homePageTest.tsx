import * as React from 'react'
import { shallow } from 'enzyme'
import HomePage from 'src/components/HomePage/HomePage'
import toJson from 'enzyme-to-json'

describe('<HomePage /> without knowing isIsolatedServer', () => {

  jest.spyOn(React, 'useContext').mockImplementation(() => ({
    isIsolatedServer: null
  }))

  const homePage = shallow(<HomePage />)

  it('matches snapshot', () => {
    expect(toJson(homePage)).toMatchSnapshot()
  })

})

describe('Isolated Server\'s <HomePage />', () => {

  jest.spyOn(React, 'useContext').mockImplementation(() => ({
    isIsolatedServer: true
  }))

  const homePage = shallow(<HomePage />)

  it('matches snapshot', () => {
    expect(toJson(homePage)).toMatchSnapshot()
  })

})

describe('Non-Isolated Server <HomePage />', () => {

  jest.spyOn(React, 'useContext').mockImplementation(() => ({
    isIsolatedServer: false
  }))

  const homePage = shallow(<HomePage />)

  it('matches snapshot', () => {
    expect(toJson(homePage)).toMatchSnapshot()
  })

})