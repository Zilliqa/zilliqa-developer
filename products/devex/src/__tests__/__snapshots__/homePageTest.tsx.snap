// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> without knowing isIsolatedServer matches snapshot 1`] = `
<Fragment>
  <div
    className="center-spinner"
  >
    <Spinner
      animation="border"
    />
  </div>
</Fragment>
`;

exports[`Isolated Server's <HomePage /> matches snapshot 1`] = `
<Fragment>
  <div>
    <Searchbar
      isHeaderSearchbar={false}
      isISSearchbar={true}
    />
    <Dashboard />
  </div>
</Fragment>
`;

exports[`Non-Isolated Server <HomePage /> matches snapshot 1`] = `
<Fragment>
  <div>
    <Searchbar
      isHeaderSearchbar={false}
      isISSearchbar={false}
    />
    <Dashboard />
  </div>
</Fragment>
`;
